### PR TITLE
Migrate Slovakia holidays to holiday groups

### PR DESCRIPTION
**Pull request by kasya**
## Proposed change

Migrate Slovakia holidays to holiday_groups.
Update Slovakia tests.

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency upgrade (version update)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good: `make pre-commit`
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/dr-prodigy/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/dr-prodigy/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: tests/countries/test_slovakia.py [hunk 1]
```diff
@@ -9,44 +9,143 @@
...
+        self.assertHolidayNameInYears(
+            "Sviatok Všetkých svätých", range(1990, 2023)
+        )
+
+    def test_day_of_freedom_and_democracy(self):
+        self.assertHoliday(f"{year}-11-17" for year in range(2001, 2023))
+        self.assertHolidayNameInYears(
+            "Deň boja za slobodu a demokraciu", range(2001, 2023)
+        )
+        self.assertNoHoliday(f"{year}-11-17" for year in range(1950, 2000))
```
Comment (KJhellico) [inline tests/countries/test_slovakia.py:111]:
```suggestion
        self.assertNoHoliday(f"{year}-11-17" for year in range(1950, 2001))
```

File: tests/countries/test_slovakia.py [hunk 2]
```diff
@@ -9,44 +9,143 @@
...
-        self.assertIn(date(2018, 10, 30), self.holidays)
+        self.assertHoliday(
+            "2018-10-30",
+        )
+
+    def test_new_years_day(self):
+        self.assertHoliday(f"{year}-01-01" for year in range(1990, 2023))
+        self.assertHolidayNameInYears(
+            "Deň vzniku Slovenskej republiky", range(1990, 2023)
+        )
```
Comment (KJhellico) [inline tests/countries/test_slovakia.py:34]:
```suggestion
        self.assertHolidaysName(
            "Deň vzniku Slovenskej republiky",
            (f"{year}-01-01" for year in range(1990, 2023)),
        )
```
Check it together by date and by name. And the same for rest of holidays.

File: tests/countries/test_slovakia.py [hunk 3]
```diff
@@ -9,44 +9,143 @@
...
+    def test_labor_day(self):
+        self.assertHoliday(f"{year}-05-01" for year in range(1990, 2023))
+        self.assertHolidayNameInYears("Sviatok práce", range(1990, 2023))
+
+    def test_day_of_victory_over_fascizm(self):
+        self.assertHoliday(f"{year}-05-08" for year in range(1997, 2023))
+        self.assertHolidayNameInYears(
+            "Deň víťazstva nad fašizmom", range(1997, 2023)
+        )
+        self.assertNoHoliday(f"{year}-05-08" for year in range(1950, 1996))
```
Comment (KJhellico) [inline tests/countries/test_slovakia.py:73]:
```suggestion
        self.assertNoHoliday(f"{year}-05-08" for year in range(1950, 1997))
```

File: tests/countries/test_slovakia.py [hunk 4]
```diff
@@ -9,44 +9,143 @@
...
+        self.assertHolidayNameInYears(
+            "Výročie Slovenského národného povstania", range(1990, 2023)
+        )
+
+    def test_constitution_day(self):
+        self.assertHoliday(f"{year}-09-01" for year in range(1992, 2023))
+        self.assertHolidayNameInYears(
+            "Deň Ústavy Slovenskej republiky", range(1992, 2023)
+        )
+        self.assertNoHoliday(f"{year}-09-01" for year in range(1950, 1991))
```
Comment (KJhellico) [inline tests/countries/test_slovakia.py:92]:
```suggestion
        self.assertNoHoliday(f"{year}-09-01" for year in range(1950, 1992))
```

**Review by KJhellico — COMMENTED** [review]
@kasya, nice to see you here again! 👍 Please look at my suggestions.

By the way, I also started editing Slovakia (migrate to holiday groups and localization), so now I'll wait for this PR merging. 🙂

**Review by KJhellico — COMMENTED** [review]
LGTM 👍

File: holidays/countries/slovakia.py [hunk 5]
```diff
@@ -31,45 +28,65 @@ class Slovakia(HolidayBase):
...
-        self[easter_date + td(days=+1)] = "Veľkonočný pondelok"
+        # Easter.
+        # Good Friday.
+        self._add_good_friday("Veľký piatok")
 
-        self[date(year, MAY, 1)] = "Sviatok práce"
+        # Easter Monday.
+        self._add_easter_monday("Veľkonočný pondelok")
 
+        # Labour Day.
```
Comment (arkid15r) [inline holidays/countries/slovakia.py:55]:
```suggestion
        # Labor Day.
```

**Review by arkid15r — COMMENTED** [review]
🇸🇰 LGTM

**Comment by arkid15r** [comment]
Merged in, thank you!

**Pull request merged**